Profesja/Include/Profesja.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+#include <iostream>
+#include <vector>
+#include <string>
+
+struct Player
+{
+  unsigned int currentHp = 20;
+  unsigned int power = 1;
+  std::vector <std::string> handdeck = {"karta1", "karta2", "karta3"};
+  class Profession
+  {
+   public:
+    virtual void useUlt(Player& p1, Player& p2) = 0;

```

Comment:
1. Nie potrzebujemy trzymać Player& p1, który zapewne oznacza nas, mamy do tego `this`
2. `p2` powinien nazywać się lepiej. Przemyśleć czy nie lepiej zrobić std::optional, ponieważ nie każdy (np. Paladin) potrzebuje go, wtedy będzie dostawał nic.

Reply:
co do 2. to narazie nie chcieliśmy używać optionala, bo mozliwe ze paladyn w przyszlosci bedzie potrzebował p2 do jakiś działań